lib/MusicBrainz/Server/Constants.pm

```diff
@@ -955,6 +957,30 @@ Readonly our $PASSPHRASE_BCRYPT_COST => 12;
 Readonly our $OAUTH_INSTALLED_APP_REDIRECT_URI_RE => qr/^(?![_-])[\w-]+(?:\.(?![_-])[\w-]+)+:/;
 Readonly our $OAUTH_WEB_APP_REDIRECT_URI_RE => qr/^https?:\/\//;
 
+=item %ALIAS_LOCALES
+
+Historically, alias locales have been stored in the database and
+returned in the web service using underscores instead of dashes, e.g.
+zh_Hant_HK instead of zh-Hant-HK. Presumably this was simply because
+that was how DateTime::Locale returned them prior to version 1.00.
+The primary reason to continue doing this is to maintain compatibility
+for data users.
```

Comment:
Given we have had both kinds forever now, wouldn't it make sense to consider compatibility broken already and just go with the canonical form?

Reply:
No because even though we've had the hyphened ones for a while now, the underscored ones are still vastly more prevalent (163709 with underscores vs. only 4135 with hyphens). Any downstream data users who've written code to use alias locales (whether from the DB or web service) would very likely see their code break if we changed that. Here's just one example from Picard: https://github.com/metabrainz/picard/blob/8245afbd5ba1f345e7ab61db83dc7af67e6f482c/picard/mbjson.py#L205

Reply:
Ouch. Ok then.